core/src/main/java/org/apache/iceberg/rest/responses/ErrorResponse.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.responses;
+
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Standard response body for all API errors
+ */
+public class ErrorResponse {
+
+  private String message;
+  private String type;
+  private Integer code;
+
+  public ErrorResponse() {
+    // Needed for Jackson deserialization
+  }
+
+  private ErrorResponse(String message, String type, int code) {
+    this.message = message;
+    this.type = type;
+    this.code = code;
+    validate();
+  }
+
+  ErrorResponse validate() {
+    Preconditions.checkArgument(code != null, "Invalid response, missing field: code");
```

Comment:
If we don't use Jackson serialization, do we need a validate method? I would assume not. If this is for consistency with the other objects, I'd just leave this blank.

Reply:
Yeah this is mostly for consistency with the other objects of type `response`.

I just have it returning `this` (which a few classes do as well).